randovania/server/multiplayer/session_admin.py

```diff
@@ -243,11 +253,13 @@ def _change_layout_description(sa: ServerApp, session: MultiplayerSession, descr
             raise error.InvalidActionError("One of the worlds has undefined order field.")
 
         try:
-            description = LayoutDescription.from_bytes(
-                description_bytes, presets=[VersionedPreset.from_str(world.preset) for world in worlds]
-            )
+            presets = [VersionedPreset.from_str(world.preset) for world in worlds]
```

Comment:
These checks and the error below are pointless.when uploading a Layout Description, it's encoded in a special binary format that excludes the presets. You can see above how they're fetched from the db.

We'd need to include some sort of hash of the preset to double check, but it means a data change sadly.
What can be done is check if the game matches though. That data is redundant and saved in both places.

Reply:
I assume, due to what already has been done, we already have incompatible presets in the database itself. So, the check is not really pointless even if it fetches it from the db?

The error below is not pointless. It's a little bit inaccurate, I admit, but if the preset and game is not matching it throws the `ValueError` from `LayoutDescription.from_bytes` like:
```
randovania\layout\game_patches_serializer.py", line 159, in decode_single
    raise ValueError(f"Expected '{game.game.value}', got '{game_modifications['game']}'")
```
and with my change it is catched.

Though it is already prevented by the changes in the randovania frontend to not even do that but I thought it is nicer to not only rely on the "frontend being nice to the backend".

Honestly, as the description is sent in a byte format, I haven't seen a nice way to check for a matching game in preset and layout, that's why I just catched the error it will throw.

Reply:
ah yes, the exception capturing is good. I was originally writing the message was wrong though but then saw the other issue

The layout is also saved in binary in the db, to save some space (99% of the db size was the jsons, and having 10+ copies of the preset was pretty wasteful) so the mismatch can be there. But we can't do much so we cry I guess.

The current api has a design error in that the server must trust the client to upload a layout whose presets match the current session, with no way to confirm.

At least checking the game matches is helpful, and the client refusing to do it also helps.

Reply:
> At least checking the game matches is helpful, and the client refusing to do it also helps.

That's what this PR (hopefully) already does.
I think I could use `LayoutDescription.from_bytes` without the presets and then compare if layout and preset match instead of calling it with the presets and catch the error it will throw.

So, you want me to change something there or leave it?

The full crazy blown up PR would probably to fix this:
> The current api has a design error in that the server must trust the client to upload a layout whose presets match the current session, with no way to confirm.

Reply:
^ @henriquegemignani 🙋‍♂️

Reply:
Calling `_check_preset_consistency` here right now is not useful. These presets are fetched from the DB so they should already be valid. Could keep the call, but can you at least add a `FIXME` note so it's clear this is wrong?

Reply:
It was intended for the fact that we already have invalid presets in the db, don't we?
The thing which brought the whole thing, is still part of the db or was it removed?
Otherwise I can of course just remove it.

Reply:
If we were to care about the existing wrong values, we'd be doing a data migration to fix. I don't really care about these few that are wrong.

Reply:
Alright. Removed it and just left the two invalid action things.